rpc_state_reader_sn_api/src/lib.rs

```diff
@@ -1211,10 +1329,32 @@ mod starknet_in_rust_transaction_tests {
 
     #[cfg(test)]
     mod test {
+        use pretty_assertions_sorted::{assert_eq, assert_eq_sorted};
         use starknet_in_rust::execution::CallInfo;
+        use test_case::test_case;
 
         use super::*;
 
+        #[test]
+        fn test_get_transaction_try_from() {
+            let rpc_state = RpcState::new(RpcChain::MainNet, BlockTag::Latest.into());
+            let str_hash = "0x5d200ef175ba15d676a68b36f7a7b72c17c17604eda4c1efc2ed5e4973e2c91";
+            let tx_hash = TransactionHash(stark_felt!(str_hash));
+
+            let sn_tx = rpc_state.get_transaction(&tx_hash);
+            match &sn_tx {
+                SNTransaction::Invoke(sn_tx) => {
+                    let tx = InvokeFunction::from_invoke_transaction(
+                        sn_tx.clone(),
+                        StarknetChainId::MainNet,
+                    )
+                    .unwrap();
+                    assert_eq!(format!("0x{}", tx.hash_value().to_str_radix(16)), str_hash)
```

Comment:
It won't affect this test but comparing numerical strings may cause false errors when the source and the target are formatted differently, for example with zero-padded numbers. I think it would be better to compare their numerical values instead.